awscli/customizations/s3/subcommands.py

```diff
@@ -1824,3 +1836,25 @@ def _validate_sse_c_copy_source_for_paths(self):
                     '--sse-c-copy-source is only supported for '
                     'copy operations.'
                 )
+
+    def _validate_no_overwrite_for_download_streaming(self):
+        """
+        Validates that no-overwrite parameter is not used with streaming downloads.
+
+        When downloading from S3 to stdout (streaming download), the no-overwrite
+        parameter doesn't make sense since stdout is not a file that can be checked
+        for existence. This method checks for this invalid combination and raises
+        an appropriate error.
```

Comment:
On Unix-like systems, stdout is treated as a file-like object and you can check its existence (eg `/dev/stdout`). I think what we're trying to say is that stdout is always going to be there, so it doesn't make sense to support the no overwrite feature